go/interpreter/lfvm/memory_test.go

```diff
@@ -43,7 +45,143 @@ func TestExpansionCosts(t *testing.T) {
 		if cost != test.cost {
-			t.Errorf("ExpansionCosts(%d) = %d, want %d", test.size, cost, test.cost)
+			t.Errorf("getExpansionCosts(%d) = %d, want %d", test.size, cost, test.cost)
 		}
 	}
 }
+
+func TestMemory_expandMemoryWithoutCharging(t *testing.T) {
+
+	test := map[string]struct {
+		size        uint64
+		initialMem  []byte
+		expectedMem []byte
+	}{
+		"empty-memory-increases-to-desired-size": {
+			size:        32,
+			initialMem:  []byte{},
+			expectedMem: []byte{31: 0x0},
+		},
+		"memory-bigger-than-size-changes-nothing": {
+			size:        32,
+			initialMem:  []byte{63: 0x0},
+			expectedMem: []byte{63: 0x0},
+		},
+		"size-zero-changes-nothing": {
+			size:        0,
+			initialMem:  []byte{},
+			expectedMem: []byte{},
+		},
+		"check-memory-increases-by-32": {
+			size:        41,
+			initialMem:  []byte{},
+			expectedMem: []byte{63: 0x0},
+		},
+	}
+
+	for name, test := range test {
+		t.Run(name, func(t *testing.T) {
+			m := NewMemory()
+			m.store = test.initialMem
+			fee := m.getExpansionCosts(test.size)
+			m.expandMemoryWithoutCharging(test.size)
+			if !bytes.Equal(m.store, test.expectedMem) {
+				t.Errorf("unexpected memory value, want: %x, got: %x", test.expectedMem, m.store)
+			}
+			if m.currentMemoryCost != fee {
+				t.Errorf("unexpected total memory cost, want: %d, got: %d", fee, m.currentMemoryCost)
+			}
+		})
+	}
+
+}
+
+func TestMemory_expandMemory_ErrorCases(t *testing.T) {
+
+	tests := map[string]struct {
+		size     uint64
+		offset   uint64
+		gas      tosca.Gas
+		expected error
+	}{
+		"not enough gas": {
+			size:     32,
+			offset:   0,
+			gas:      0,
+			expected: errOutOfGas,
+		},
+		"offset overflow": {
+			size:     1,
+			offset:   math.MaxUint64,
+			gas:      100,
+			expected: errGasUintOverflow,
+		},
+		"size overflow": {
+			size:     math.MaxUint64,
+			offset:   1,
+			gas:      100,
+			expected: errGasUintOverflow,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			ctxt.gas = test.gas
+
+			err := m.expandMemory(test.offset, test.size, &ctxt)
+			if !errors.Is(err, test.expected) {
+				t.Errorf("unexpected error, want: %v, got: %v", test.expected, err)
+			}
+			if m.length() != 0 {
+				t.Errorf("should have remained empty, got length: %d", m.length())
+			}
+		})
+	}
+}
+
+func TestMemory_expandMemory_expandsMemoryOnlyWhenNeeded(t *testing.T) {
+
+	tests := map[string]struct {
+		size               uint64
+		offset             uint64
+		initialMemory      []byte
+		expectedMemorySize uint64
+	}{
+		"empty-memory-with-zero-offset-and-size-does-not-expand": {},
+		"size-zero-with-offset-does-not-expand": {
+			size:               0,
+			offset:             32,
+			expectedMemorySize: 0,
+		},
+		"expand-memory-in-words-length": {
+			size:               13,
+			offset:             0,
+			expectedMemorySize: 32,
+		},
+		"memory-bigger-than-size+offset-does-not-expand": {
```

Comment:
nit: when starting to mixing `-` and `+` it might be better to use spaces for word separation